charms/metallb-controller/src/charm.py

```diff
@@ -21,93 +21,84 @@
 logger = logging.getLogger(__name__)
 
 
-class MetallbControllerCharm(CharmBase):
+class MetalLBControllerCharm(CharmBase):
     """MetalLB Controller Charm."""
 
     _stored = StoredState()
 
     def __init__(self, *args):
         """Charm initialization for events observation."""
         super().__init__(*args)
-        if not self.model.unit.is_leader():
-            self.model.unit.status = WaitingStatus("Waiting for leadership")
+        if not self.unit.is_leader():
+            self.unit.status = WaitingStatus("Waiting for leadership")
             return
         self.image = OCIImageResource(self, 'metallb-controller-image')
-        self.framework.observe(self.on.start, self.on_start)
+        self.framework.observe(self.on.install, self._on_start)
```

Comment:
Ah, we definitely should include `leader-elected`.  In theory, the first unit should have leadership from the get-go because there's no other unit to have claimed it already.  I would consider it not having everything it needs during the install hook to be a bug, just like [sometimes not having resources](https://bugs.launchpad.net/juju/+bug/1763175) or [the install hook not being called at all](https://discourse.juju.is/t/install-hook-now-called-for-kubernetes-charms/2585/2).  So, while technically it most likely won't do anything during the install hook, I was including that because in an ideal world that would lead to the fastest startup time for the workload.

Reply:
Oh, I should also note that if the charm doesn't have leadership, then the observe won't even be called, so it will be exactly the same as not having it there (no extra work done) but including it allows us the possibility to get the workload up sooner if there's some chance that we can.